### ISSUE TITLE
There are a TON of warnings

### ISSUE THREAD

**Issue by Elronnd** [thebracket/bgame#126]
If you want, I can try to clean some of them up and make a PR, but there are a ***TON*** of them

**Comment by thebracket** [comment]
You're absolutely right - there are a ton of warnings (depending upon which compiler you use, there are even more - GCC gives a lot more than Clang). My goal is to do a big cleanup on that (and some global variable issues) soon, but I want to stabilize some architecture decisions first. Thanks!

**Comment by thebracket** [comment]
Two changes to RLTK just went up that massively reduced the number of warnings (especially on Clang). I'd omitted some braces in an initializer list (it still worked, oddly), in `each` and `each_if` - which are used _everywhere_. That's a good start!

**Comment by thebracket** [comment]
I just went on a big warning hunt, and removed a LOT of warnings from Clang builds. I'll have to do the same for GCC, but not today.
